webapp/src/store/theme.ts

```diff
@@ -0,0 +1,17 @@
+import { defineStore } from "pinia";
+import { ref } from 'vue'
+
+
+export const useThemeStore = defineStore("theme", () => {
+  const currentTheme = ref<"light" | "dark">("light");
+  
+  function toggleTheme() {
+    const newTheme = currentTheme.value === "light" ? "dark" : "light";
+    currentTheme.value = newTheme;
+  }
+  return { currentTheme, toggleTheme }
+}, {
+  persistedState: {
+    persist: true
+  }
+});
```

Comment:
small rewrite of store
 * mv getColor direclty here
   * return computed ref instead of using `watch`
 * `currentTheme` -> `current`
   * avoid the double "theme" when `useThemeStore().currentTheme`
 * inline `toogleTheme` as used only once
 
```suggestion
import { defineStore } from "pinia";
import type { Ref } from "vue";
import { computed, ref } from "vue";

export const useThemeStore = defineStore(
	"theme",
	() => {
		const current = ref<"light" | "dark">("light");

		function getColor(lightColor: string, darkColor: string): Ref<string> {
			return computed(() =>
				current.value === "light" ? lightColor : darkColor,
			);
		}

		return { current, getColor };
	},
	{ persistedState: { persist: true } },
);
```

Reply:
Very nice!